qiskit_nature/problems/second_quantization/lattice/lattice/hyper_cubic_lattice.py

```diff
@@ -0,0 +1,326 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The hyper-cubic lattice"""
+from dataclasses import asdict
+from itertools import product
+from math import pi
+from typing import Dict, List, Optional, Tuple, Union
+import numpy as np
+from retworkx import PyGraph
+
+from .lattice import DrawStyle, Lattice
+from .boundary_condition import BoundaryCondition
+
+
+def _coordinate_to_index(coord: np.ndarray, size: Tuple[int, ...]) -> int:
+    """Convert the coordinate of a lattice point to an integer for labeling.
+        When size=(l0, l1, l2, ...), then a coordinate (x0, x1, x2, ...) is converted as
+        x0 + x1*l0 + x2*l0*l1 + ... .
+    Args:
+        coord: Input coordinate to be converted.
+        size: Lengths of each dimension.
+
+    Returns:
+        int: Return x0 + x1*l0 + x2*l0*l1 + ...
+            when coord=np.array([x0, x1, x2...]) and size=(l0, l1, l2, ...).
+    """
+    dim = len(size)
+    base = np.array([np.prod(size[:i]) for i in range(dim)], dtype=int)
+    return np.dot(coord, base).item()
+
+
+def _self_loops(size: Tuple[int, ...], onsite_parameter: complex) -> List[Tuple[int, int, complex]]:
```

Comment:
I like that you have extracted this logic into standalone functions but why did you not make these functions of the `HyperCubicLattice` class? That would avoid having to pass in parameters like `onsite_parameter`, `edge_parameter`, etc. which are all available on `self` of a `HyperCubicLattice` instance.

Reply:
Thanks for the comment. You mean to write them as functions belonging to tha `HyperCubicLattice` class, not global functions, write? It sounds good to me. 
On the other hand, these functions are used in the initialization function `__init__()`, so I wonder if it is possible (or appropriate) to write them as functions of the class.

Reply:
I don't see why that should cause a problem. Before the class will be instantiated all the functions will already be available. You will just need to ensure to set the attributes which your accessing on `self` before calling these functions.

Reply:
Thanks for the reply. I was wrong. I have included these functions in the class `HyperCubicLattice`.